### PR TITLE
Provide support for scraping Azure Key Vault

### LINKED ISSUE #311: Provide support for scraping Azure Key Vault
Provide support for scraping metrics from Azure Key Vault

### PR DESCRIPTION
**Pull request by ChristianEder**
Fixes #311

### CODE REVIEW

**Comment by ChristianEder** [comment]
Metrics are provided locally:

\# HELP azure_kv_latency Latency
\# TYPE azure_kv_latency gauge
azure_kv_latency{resource_group="ched",subscription_id="c24aae66-cc75-4de5-8148-c83a5981c88f",resource_uri="subscriptions/c24aae66-cc75-4de5-8148-c83a5981c88f/resourceGroups/ched/providers/Microsoft.KeyVault/vaults/mypromitorkv",instance_name="mypromitorkv"} 343
\# HELP promitor_ratelimit_arm Indication how many calls are still available before Azure Resource Manager is going to throttle us.
\# TYPE promitor_ratelimit_arm gauge
promitor_ratelimit_arm{tenant_id="1a7dacf1-7fd4-45b1-bf70-01265d702d04",subscription_id="c24aae66-cc75-4de5-8148-c83a5981c88f",app_id="9d78938a-034b-429f-ae35-c57b969e080f"} 11998

File: docs/configuration/v1.x/metrics/key-vault.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+## Azure Key Vault - ![Availability Badge](https://img.shields.io/badge/Available%20Starting-v1.6-green.svg)
+
+You can declare to scrape an Azure Key Vault
+via the `KeyVault` resource type.
+
+The following fields need to be provided:
+
+- `keyVaultName` - The name of the Azure Key Vault
```
[the remark this answers is not in the document]
Comment (ChristianEder) [reply docs/configuration/v1.x/metrics/key-vault.md]: done

**Comment by ChristianEder** [comment]
You made this easy to extend :-)

One question: would dedicated scrapers for IoT Hub, Key Vault etc have been neccessary, or could they have been done with the generic scraper? What is the exact benefit of a dedicated scraper? That users dont have to build the resource type specific resource identifier, including the namespace such as "Microsoft.KeyVault/vaults"?

**Comment by tomkerkhove** [comment]
That's great to hear!

You can scrape every single resource today with the generic one but these are more of "convenient" scrapers where we take the information and do all the magic for you! More of a user-friendly way of scraping let's say.

**Pull request merged**